### DISCUSSION TITLE
请问有没有服务或者方法或者当前运行环境、版本号？

### DISCUSSION THREAD

**Discussion by lovelytan** [opensumi/core#2412] [Q&A] [answered]
项目有web版和Electron版本，复用自定义埋点上报。判断当前的**运行环境**（web还是桌面版）、**版本号**，除了useragent，webpack变量啥的，openSumi有没有封装的判断方法或服务。

**Comment by erha19** [comment, marked as answer]
@naduohua 如果是前端环境，建议采用下面的方式去判断
```ts

@Autowired(AppConfig)
private appConfig: AppConfig;

this.appConfig.isElectronRenderer // 用这个值做判断
```

Node 进程一般在功能实现上不需要区分。

版本号这块，建议是通过 `webpack.DefinePlugin` 插件去通过读取依赖的 `@opensumi/ide-core-common/pakcage.json` 包中的版本号
